apps/core/components/searchForm/SearchForm.js

```diff
@@ -0,0 +1,231 @@
+// @flow
+
+import * as React from 'react';
+import { View, Platform } from 'react-native';
+import { defaultTokens } from '@kiwicom/orbit-design-tokens';
+import {
+  LAYOUT,
+  withLayoutContext,
+  type LayoutContextState,
+} from '@kiwicom/margarita-device';
+import { StyleSheet, Button, Icon } from '@kiwicom/universal-components';
+import { BASIC_ISO_DATE_FORMAT, TRIP_TYPES } from '@kiwicom/margarita-config';
+import format from 'date-fns/format';
+import {
+  withNavigation,
+  Routes,
+  type Navigation,
+} from '@kiwicom/margarita-navigation';
+import {
+  withAlertContext,
+  type AlertContent,
+  type AlertContextState,
+} from '@kiwicom/margarita-components';
+
+import {
+  withSearchContext,
+  type SearchContextState,
+  type Location,
+} from '../../scenes/search/SearchContext';
+import Placepickers from './Placepickers';
+import Datepickers from './Datepickers';
+import SearchModal from './SearchModal';
+import SearchFormModes from './SearchFormModes';
+
+type SearchParameters = {|
+  +dateFrom: string,
+  +dateTo: string,
+  +returnDateFrom: string,
+  +returnDateTo: string,
+  +tripType: string,
+  +travelFrom: ?$ReadOnlyArray<Location>,
+  +travelTo: ?$ReadOnlyArray<Location>,
+  +sort: string,
+  +adults: number,
+  +infants: number,
+  +bags: number,
+|};
+
+type Props = {
+  +navigation: Navigation,
+  +travelFrom: ?$ReadOnlyArray<Location>,
+  +travelTo: ?$ReadOnlyArray<Location>,
+  +dateFrom: Date,
+  +dateTo: Date,
+  +returnDateFrom: Date,
+  +returnDateTo: Date,
+  +tripType: string,
+  +adults: number,
+  +infants: number,
+  +bags: number,
+  +layout: number,
+  +setAlertContent: (alertContent: AlertContent | null) => void,
+  +onSubmit?: SearchParameters => void,
+};
+
+class SearchForm extends React.Component<Props> {
+  convertLocationsToParams = (locations, field) => {
+    if (Array.isArray(locations)) {
+      return locations.reduce((acc, location, index) => {
+        const prefix = index > 0 ? ',' : '';
+        const locationField = location[field] || '';
+        return `${acc}${prefix}${locationField}`;
+      }, '');
+    }
+    return '';
+  };
+
+  handleSubmitPress = () => {
+    const {
+      travelFrom,
+      travelTo,
+      tripType,
+      adults,
+      infants,
+      bags,
+      onSubmit,
+    } = this.props;
+    if (travelFrom == null) {
```

Comment:
We broke this condition with some other update. But I just spotted it. 
Can you please extend it to `travelFrom == null || travelFrom.length === 0`. The `travelFrom` is now array (and can be empty) it was just single object before.